clients/roscpp/src/libros/this_node.cpp

```diff
@@ -89,6 +89,12 @@ void init(const std::string& name, const M_string& remappings, uint32_t options)
 #endif
   }
 
+  if (name.empty()) {
+    std::stringstream ss;
+    ss << "Name [" << name << "] must not be empty";
```

Comment:
Since the name is empty anyway (as checked two lines above) there is no need to build the message using stringstream. `throw InvalidNameException("The node name must not be empty")` should be sufficient.